src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,50 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T, W> {
+    private T first;
+    private W second;
 
+    private Pair(T first, W second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public T getFirst() {
+        return first;
+    }
+
+    public W getSecond() {
+        return second;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 17;
+        result = 21 * result + (first == null ? 0 : first.hashCode());
+        result = 21 * result + (second == null ? 0 : second.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object pair) {
+        if (pair == this) {
+            return true;
+        }
+        if (pair == null) {
+            return false;
+        }
+        if (pair.getClass().equals(Pair.class)) {
+            Pair current = (Pair) pair;
```

Comment:
let's use parameterisation here
we should avoid using raw types

Reply:
done